web/src/components/MemoContent/Link.tsx

```diff
@@ -1,18 +1,92 @@
+import { Tooltip, Card, AspectRatio, Box } from "@mui/joy";
+import { Link as MLink } from "@mui/joy";
+import { useEffect, useState } from "react";
+import { metadataServiceClient } from "@/grpcweb";
+import { LinkMetadata } from "@/types/proto/api/v2/link_service";
+
 interface Props {
   url: string;
   text?: string;
 }
 
 const Link: React.FC<Props> = ({ text, url }: Props) => {
+  const [linkMetadata, setLinkMetadata] = useState<LinkMetadata | undefined>();
+
+  const fetchUrlMetadata = async () => {
+    try {
+      const response = await metadataServiceClient.getLinkMetadata({ link: url }, {});
+      setLinkMetadata(response.metadata);
+    } catch (error) {
+      console.error("Error fetching URL metadata:", error);
+      return null;
+    }
+  };
+
+  useEffect(() => {
+    fetchUrlMetadata();
+  }, [url]);
+
   return (
-    <a
-      className="text-blue-600 dark:text-blue-400 cursor-pointer underline break-all hover:opacity-80 decoration-1"
-      href={url}
-      target="_blank"
-      rel="noopener noreferrer"
-    >
-      {text || url}
-    </a>
+    <>
+      {/* Small screens */}
+      <div className="flex md:hidden">
+        <MLink href={url} underline="none" sx={{ fontWeight: "lg" }}>
+          {url || text}
+        </MLink>
+      </div>
+
+      {/* Medium & Above screens */}
```

Comment:
We can use `useResponsiveWidth` to conditional rendering tooltip.
https://github.com/usememos/memos/blob/main/web/src/hooks/useResponsiveWidth.ts#L9

Reply:
In small devices, we should not render the link card _physically_.

Reply:
Yes sorry my bad I understand now how to use `useResponsiveWidth` in the code